src/main/java/net/mrscauthd/beyond_earth/events/CoalTorchEvents.java

```diff
@@ -24,49 +25,45 @@
     public static void onBlockPlace(BlockEvent event) {
         Level world = (Level) event.getWorld();
 
-        double x = event.getPos().getX();
-        double y = event.getPos().getY();
-        double z = event.getPos().getZ();
+        final double x = event.getPos().getX();
+        final double y = event.getPos().getY();
+        final double z = event.getPos().getZ();
 
-        BlockPos pos = new BlockPos(x,y,z);
+        final BlockPos pos = new BlockPos(x, y, z);
 
         if (Methods.isSpaceWorldWithoutOxygen(world)) {
 
+            final BlockState blockState = world.getBlockState(pos);
+            final Block block = blockState.getBlock();
```

Comment:
Rather than calling `.getBlockState(pos).getBlock()` multiple times, call it once, store the result in a variable called `block` and use the variable instead.

Grabbing from variables is faster than calling methods with parameters (a parameter being `pos` in this case)